ArjunaCore/arjuna/tests/classes/com/hp/mwtests/ts/arjuna/recovery/BasicActionExpiryScannerTest.java

```diff
@@ -0,0 +1,129 @@
+        recoveryPropertyManager.getRecoveryEnvironmentBean().setExpiryScanInterval(1);
+    }
+
+    public OutputObjectState createObjectState(Uid txId, Uid processUid) throws IOException {
+        OutputObjectState os = new OutputObjectState();
+        os.packStringBytes("#ARJUNA#".getBytes(StandardCharsets.UTF_8));
+        UidHelper.packInto(txId, os);
+        UidHelper.packInto(processUid, os);
+        os.packLong(Instant.now().minus(Duration.ofHours(1)).toEpochMilli());
+        os.packBoolean(true);
+        os.packInt(RecordType.NONE_RECORD);
+        os.packInt(0);
+        os.packInt(ActionStatus.H_HAZARD);
+        os.packInt(ActionType.TOP_LEVEL);
+        os.packInt(TwoPhaseOutcome.HEURISTIC_HAZARD);
+        return os;
+    }
+
+    @Test
+    public void testAdvancedAtomicActionExpiryScanner() {
+        boolean passed = false;
+        Uid txId = new Uid();
+        Uid processUid = new Uid();
+
+        try {
+            OutputObjectState os = createObjectState(txId, processUid);
+
+            log.debug("Creating dummy log");
+            recoveryStore.write_committed(txId, TYPE_NAME, os);
+            if (recoveryStore.currentState(txId, TYPE_NAME) == StateStatus.OS_COMMITTED) {
+                log.debugf("Wrote dummy transaction %n", txId);
+            }
+
+            AdvancedAtomicActionExpiryScanner scanner = new AdvancedAtomicActionExpiryScanner();
+            scanner.scan();
+
+            if (recoveryStore.currentState(txId, TYPE_NAME + "/Expired") == StateStatus.OS_COMMITTED) {
+                log.debug("Transaction log moved!");
+                passed = true;
+            } else {
+                log.debug("Transaction log not moved!");
+            }
+        } catch (Exception ex) {
+            ex.printStackTrace();
+        }
+
+        assertTrue(passed);
+    }
+
+    @Test
+    public void testAdvancedAtomicActionPurgeExpiryScanner() {
+        boolean passed = false;
+        Uid txId = new Uid();
+        Uid processUid = new Uid();
+
+        try {
+            OutputObjectState os = createObjectState(txId, processUid);
+
+            log.debug("Creating dummy log");
+            recoveryStore.write_committed(txId, TYPE_NAME, os);
+            if (recoveryStore.currentState(txId, TYPE_NAME) == StateStatus.OS_COMMITTED) {
+                log.debugf("Wrote dummy transaction %n", txId);
+            }
+
+            AdvancedAtomicActionPurgeExpiryScanner scanner = new AdvancedAtomicActionPurgeExpiryScanner();
+            scanner.scan();
+
+            if (recoveryStore.currentState(txId, TYPE_NAME) == StateStatus.OS_UNKNOWN
+                    && recoveryStore.currentState(txId, TYPE_NAME + "/Expired") == StateStatus.OS_UNKNOWN) {
+                log.debug("Transaction log purged!");
+                passed = true;
+            } else {
+                log.debug("Transaction log not purged!");
+            }
+        } catch (Exception ex) {
+            ex.printStackTrace();
```

Comment:
My IDE suggests that calls to 'printStackTrace()' should probably be replaced with more robust logging. 
We have [issue JBTM-2880](https://issues.redhat.com/browse/JBTM-2880) which asks for all such occurrences to be replaced. If you have time it will help us when fixing JBTM-2880 if you could address it here.

Reply:
Replace `ex.printStackTrace();` with logger